pkgs/applications/audio/easyabc/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, fetchFromGitHub, python39, makeWrapper, fluidsynth, soundfont-fluid, wrapGAppsHook, abcmidi, abcm2ps, ghostscript }:
+
+# requires python39 due to https://stackoverflow.com/a/71902541
+python39.pkgs.buildPythonApplication rec {
+  pname = "easyabc";
+  version = "1.3.8.6";
+
+  src = fetchFromGitHub {
+    owner = "jwdj";
+    repo = pname;
+    rev = "6461b2c14280cb64224fc5299c31cfeef9b7d43c";
+    sha256 = "sha256-leC3A4HQMeJNeZXArb3YAYr2mddGPcws618NrRh2Q1Y=";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  propagatedBuildInputs = with python39.pkgs; [
+    cx_Freeze
+    wxPython_4_2
+    pygame
+  ];
+
+  # apparently setup.py only supports Windows and Darwin
+  # everything is very non-standard in this project
+  dontBuild = true;
+  format = "other";
+
+  # https://discourse.nixos.org/t/packaging-mcomix3-python-gtk-missing-gsettings-schemas-issue/10190/2
+  strictDeps = false;
+
+  installPhase = ''
+    mkdir -p $out/share/easyabc
+    mv * $out/share/easyabc
+
+    sed -i '46 i\lib_locations = ["${fluidsynth}/lib/libfluidsynth.so"]' $out/share/easyabc/fluidsynth.py
+    sed -i '3962 i\        default_soundfont_path = "${soundfont-fluid}/share/soundfonts/FluidR3_GM2-2.sf2"' $out/share/easyabc/easy_abc.py
+    sed -i '8371 i\            settings["gs_path"] = "${ghostscript}/bin/gs"' $out/share/easyabc/easy_abc.py
```

Comment:
Use a patch file.

Reply:
You mean `patches = [ ... ]`. AFAIK that's for static patches, but I need to interpolate packages here?

Reply:
These magic numbers in those sed commands are too magical.

A patch file is better because if the upstream somehow modifies the file you are `sed`ing, sed will not warn you about this, generating heisenbugs.

About interpolation, you can use `substituteAll`, that substitutes `@pattern@` templates.
You can look at the code at `pkgs/shells/zsh/zsh-powerlevel10k/` as an example.

https://github.com/NixOS/nixpkgs/blob/515819f138530305a60faf15c912a640da984b89/pkgs/shells/zsh/zsh-powerlevel10k/default.nix#L34-L39

https://github.com/NixOS/nixpkgs/blob/515819f138530305a60faf15c912a640da984b89/pkgs/shells/zsh/zsh-powerlevel10k/gitstatusd.patch#L9-L10

Reply:
Thanks I know what a patch is, I just didn't know about the `substituteAll` function :+1:

Reply:
Applied in https://github.com/NixOS/nixpkgs/pull/237197/commits/5fa3e97bef6ce3b29331a1d0cfc006132d3c79ed